lib/ably/modules/conversions.rb

```diff
@@ -115,5 +115,38 @@ def ensure_supported_payload(payload)
 
       raise Ably::Exceptions::UnsupportedDataType.new('Invalid data payload', 400, Ably::Exceptions::Codes::INVALID_MESSAGE_DATA_OR_ENCODING)
     end
+
+    # It converts the name, data, attributes into the array of Message objects
+    #
+    # @return [Array<Ably::Models::Message>]
+    #
+    def build_messages(name, data = nil, attributes = {})
+      if name.kind_of?(Enumerable)
+        return name.map { |item| Ably::Models::Message(ensure_supported_name_and_payload(item, data, attributes).dup) }
+      end
+      [Ably::Models::Message(ensure_supported_name_and_payload(name, data, attributes).dup)]
+    end
+
+    # Ensures if the first argument (name) is a String, Hash or Ably::Models::Message object,
+    # second argument (data) should be a String, Hash, Array or nil (see ensure_supported_payload() method).
+    #
+    # @return [Hash]        It contains :name, :data and other attributes
+    #
+    # (RSL1a, RSL1b)
+    #
+    def ensure_supported_name_and_payload(name, _data = nil, attributes = {})
```

Comment:
variable with an underscore by convention is used for unused variables. It's quite confusing to see `_data` being used later on

Reply:
Fixed